crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -1552,6 +1599,40 @@ mod test {
         assert!(errors.is_empty());
     }
 
+    #[test]
+    fn resolve_fmt_strings() {
+        let src = r#"
+            fn main(x : Field) {
+                let y = println("this is the variable passed to main: {x}");
+                assert(y == "this is the variable passed to main: {x}");
+                let string = "this is x: {x}";
+                println(string);
```

Comment:
None of these lines use format strings

Reply:
I was trying to test before I add the `f` prefix that the variable ident would be unused in different circumstances but this has since changed. I have updated the test to check for a `VariableNotDeclared`